pkgs/development/interpreters/python/python-packages-base.nix

```
@@ -13,72 +14,127 @@ let
+      in
+      if builtins.isAttrs result then
+        result
+      else if builtins.isFunction result then
+        {
+          overridePythonAttrs = newArgs: makeOverridablePythonPackage f (overrideWith newArgs);
+          __functor = self: result;
+        }
+      else
+        result
+    );
+
+  mkPythonDerivation =
+    if python.isPy3k then ./mk-python-derivation.nix else ./python2/mk-python-derivation.nix;
+
+  buildPythonPackage = makeOverridablePythonPackage (
+    lib.makeOverridable (
+      callPackage mkPythonDerivation {
+        inherit namePrefix; # We want Python libraries to be named like e.g. "python3.6-${name}"
+        inherit toPythonModule; # Libraries provide modules
       }
-      else result);
-
-  mkPythonDerivation = if python.isPy3k then
-    ./mk-python-derivation.nix
-  else
-    ./python2/mk-python-derivation.nix;
-
-  buildPythonPackage = makeOverridablePythonPackage (lib.makeOverridable (callPackage mkPythonDerivation {
-    inherit namePrefix;     # We want Python libraries to be named like e.g. "python3.6-${name}"
-    inherit toPythonModule; # Libraries provide modules
-  }));
-
-  buildPythonApplication = makeOverridablePythonPackage (lib.makeOverridable (callPackage mkPythonDerivation {
-    namePrefix = "";        # Python applications should not have any prefix
-    toPythonModule = x: x;  # Application does not provide modules.
-  }));
+    )
+  );
+
+  buildPythonApplication = makeOverridablePythonPackage (
+    lib.makeOverridable (
+      callPackage mkPythonDerivation {
+        namePrefix = ""; # Python applications should not have any prefix
+        toPythonModule = x: x; # Application does not provide modules.
+      }
+    )
+  );
 
   # Check whether a derivation provides a Python module.
-  hasPythonModule = drv: drv?pythonModule && drv.pythonModule == python;
+  hasPythonModule = drv: drv ? pythonModule && drv.pythonModule == python;
 
   # Get list of required Python modules given a list of derivations.
-  requiredPythonModules = drvs: let
-    modules = lib.filter hasPythonModule drvs;
-  in lib.unique ([python] ++ modules ++ lib.concatLists (lib.catAttrs "requiredPythonModules" modules));
+  requiredPythonModules =
+    drvs:
+    let
+      modules = lib.filter hasPythonModule drvs;
+    in
+    lib.unique (
+      [ python ] ++ modules ++ lib.concatLists (lib.catAttrs "requiredPythonModules" modules)
+    );
 
   # Create a PYTHONPATH from a list of derivations. This function recurses into the items to find derivations
   # providing Python modules.
   makePythonPath = drvs: lib.makeSearchPath python.sitePackages (requiredPythonModules drvs);
 
   removePythonPrefix = lib.removePrefix namePrefix;
 
-  # Convert derivation to a Python module.
-  toPythonModule = drv:
-    drv.overrideAttrs( oldAttrs: {
+  /**
+    A function that takes a python package and returns a new derivation that can be used as a python module
```

```suggestion
    A function that takes a Python package and returns a new derivation that can be used as a Python module
```

Please use the upstream spelling (and capitalisation) of project names.